src/Symfony/Component/Config/Definition/EnumNode.php

```diff
@@ -78,6 +84,10 @@ protected function finalizeValue(mixed $value): mixed
     {
         $value = parent::finalizeValue($value);
 
+        if ($this->enumFqcn && !$value instanceof \BackedEnum) {
+            $value = $this->enumFqcn::tryFrom($value);
```

Comment:
Warning: This call can produce a `TypeError` on invalid input, for instance when the user provided a boolean value.

Reply:
> then any scalar value will be casted to a string.

A gentle reminder that integer-backed enums exist. 😉

Reply:
Yep, taken into account in the last update :+1: